File: src/main/java/dev/rono/bungeechat/listeners/PlayerChatEvent.java
package dev.rono.bungeechat.listeners;

import dev.rono.bungeechat.BungeeChat;
import dev.rono.bungeechat.commands.ChatCommand;
import net.md_5.bungee.api.CommandSender;
import net.md_5.bungee.api.connection.ProxiedPlayer;
import net.md_5.bungee.api.event.ChatEvent;
import net.md_5.bungee.api.plugin.Listener;
import net.md_5.bungee.event.EventHandler;

public class PlayerChatEvent implements Listener {

    @EventHandler
    public void onChat(ChatEvent e) {
        ProxiedPlayer player = (ProxiedPlayer) e.getSender();

        if (e.getMessage().startsWith("/")) {
            return;
        }

        for (ChatCommand command : BungeeChat.commands) {
            if (command.toggledPlayers.contains(player)) {
                command.execute((CommandSender) e.getSender(), e.getMessage().split(" "));
                e.setCancelled(true);
            }
        }
    }
}

File: velocity/src/main/kotlin/dev/rono/proxychat/velocity/listeners/PlayerChatEvent.kt
package dev.rono.proxychat.velocity.listeners

import com.velocitypowered.api.event.Subscribe
import com.velocitypowered.api.event.player.PlayerChatEvent
import dev.rono.proxychat.velocity.ProxyChatVelocity

class PlayerChatEvent {
    @Subscribe
    fun onChat(event: PlayerChatEvent) {
        val player = event.player
        val message = event.message

        if (message.startsWith("/")) {
            return
        }

        for (command in ProxyChatVelocity.instance.commands) {
            if (command.useCommandPrefix && message.startsWith(command.commandPrefix) && player.hasPermission(command.chatConfig.getString("permission") ?: "")) {
                val commandMessage = message.substring(command.commandPrefix.length).trim()
                command.handleChat(player, commandMessage.split(" ").toTypedArray())
                event.result = PlayerChatEvent.ChatResult.message("/proxychat")
                return
            } else if (command.toggleUtils.isToggled(player.uniqueId)) {
                command.handleChat(player, message.split(" ").toTypedArray())
                event.result = PlayerChatEvent.ChatResult.message("/proxychat")
                return
            }
        }
    }
}
